### PR TITLE
Add badges for gradle check, gha workflow, documentation & forum

### LINKED ISSUE #1323: Add status badge for showing code coverage in README
**Is your feature request related to a problem? Please describe.**
A part of issue: #850
Depends on issue: #1320
We need a quick insight for code coverage of the repository, as a result of code coverage reporting.

**Describe the solution you'd like**
Add a [Codecov status badge](https://about.codecov.io/product/feature/badges/) to reflect the percentage of code coverage of the repository in [README.md](https://github.com/opensearch-project/OpenSearch/blob/d46c206f29c760cbe911d1ee8b51ed501a84d14e/README.md).
It has been in the [plan](https://github.com/opensearch-project/opensearch-plugins/blob/511f8d2f104787515fb7a73337a2fb7acebc7531/TESTING.md#code-coverage-reporting) for OpenSearch/Dashboards plugins, and we can expend it to OpenSearch.

Steps for adding the badge into README.md: [the guide in opensearch-plugins repo](https://github.com/opensearch-project/opensearch-plugins/blob/511f8d2f104787515fb7a73337a2fb7acebc7531/TESTING.md#a-status-badge-in-readme-file-to-show-the-code-coverage)

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by dreamer-89**
Signed-off-by: Suraj Singh <surajrider@gmail.com>

### Description
Adds badges to readme for different jobs listed below

1. GHA gradle check
2. GHA precommit
3. GHA validate pull request
4. Public jenkins gradle check
5. Documentation
6. Forum

### Issues Resolved
#1323 

### Related
#850 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
Screen Shot 2022-07-12 at 10 08 12 AM

**Comment by dreamer-89** [comment]
Unrelated test failure. Retrying job.
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.indices.replication.SegmentReplicationTargetServiceTests.testShardAlreadyReplicating" -Dtests.seed=9A5D3217135E5052 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=th-TH -Dtests.timezone=Europe/Vatican -Druntime.java=17

org.opensearch.indices.replication.SegmentReplicationTargetServiceTests > testShardAlreadyReplicating FAILED
    java.lang.AssertionError: can't move replication to stage [DONE]. current stage: [INIT] (expected [REPLICATING])
        at __randomizedtesting.SeedInfo.seed([9A5D3217135E5052:B984B70C9999995B]:0)
        at org.opensearch.indices.replication.SegmentReplicationState.validateAndSetStage(SegmentReplicationState.java:88)
        at org.opensearch.indices.replication.SegmentReplicationState.setStage(SegmentReplicationState.java:107)
        at org.opensearch.indices.replication.SegmentReplicationTarget.onDone(SegmentReplicationTarget.java:87)
        at org.opensearch.indices.replication.common.ReplicationTarget.markAsDone(ReplicationTarget.java:143)
        at 
```

File: README.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 [![codecov](https://codecov.io/gh/opensearch-project/OpenSearch/branch/main/graph/badge.svg)](https://codecov.io/gh/opensearch-project/OpenSearch)
+[![GHA gradle check](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml)
+[![GHA validate pull request](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml)
+[![GHA precommit](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml)
+[![Jenkins gradle check job](https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fbuild.ci.opensearch.org%2Fjob%2Fgradle-check%2F&label=Jenkins%20Gradle%20Check)](https://build.ci.opensearch.org/job/gradle-check/)
```
Comment (owaiskazi19) [inline README.md:5]: This is great @dreamer-89  Just a small question, do we need to have this when we have GHA gradle check?

Comment (dreamer-89) [reply README.md:5]: Thanks @owaiskazi19 for the review. I think in general, having more info is fine. The public jenkins link here provides saves one hop to jenkins server. Without this, user need to first go to GHA action and then figure out the build number and then go to jenkins.

**Review by kotwanikunal — APPROVED** [review]

**Pull request merged**